protocols/v2/noise-sv2/src/initiator.rs

```diff
@@ -119,23 +204,36 @@ impl Initiator {
         Box::new(self_)
     }
 
-    /// #### 4.5.1.1 Initiator
+    /// Creates a new [`Initiator`] instance using a raw 32-byte public key.
     ///
-    /// Initiator generates ephemeral keypair and sends the public key to the responder:
+    /// Constructs a [`XOnlyPublicKey`] from the provided raw key slice and initializes a new
+    /// [`Initiator`] with the derived public key. If the provided key cannot be converted into a
+    /// valid [`XOnlyPublicKey`], an [`Error::InvalidRawPublicKey`] error is returned.
     ///
-    /// 1. initializes empty output buffer
-    /// 2. generates ephemeral keypair `e`, appends `e.public_key` to the buffer (64 bytes plaintext public key encoded with ElligatorSwift)
-    /// 3. calls `MixHash(e.public_key)`
-    /// 4. calls `EncryptAndHash()` with empty payload and appends the ciphertext to the buffer (note that _k_ is empty at this point, so this effectively reduces down to `MixHash()` on empty data)
-    /// 5. submits the buffer for sending to the responder in the following format
-    ///
-    /// ##### Ephemeral public key message:
+    /// Typically used when the initiator is aware of the responder's public key in advance.
+    pub fn from_raw_k(key: [u8; 32]) -> Result<Box<Self>, Error> {
+        let pk =
+            secp256k1::XOnlyPublicKey::from_slice(&key).map_err(|_| Error::InvalidRawPublicKey)?;
+        Ok(Self::new(Some(pk)))
+    }
+
+    /// Creates a new [`Initiator`] without specifying a responder's public key.
+    /// This function initializes the [`Initiator`] with a default empty state.
+    pub fn without_pk() -> Result<Box<Self>, Error> {
+        Ok(Self::new(None))
+    }
```

Comment:
we should elaborate more here to let the user know under which circumstances this could be useful

according to @Fi3 via https://github.com/stratum-mining/stratum/discussions/916#discussioncomment-10698258:

> we do not check the certificate. So you still have an encrypted connection but you don't authenticate the upstream

> For example when you connect to a TP that you run locally

Reply:
This are used when we want encrypted connection but without any authentication.

Reply:
we should rewrite the comment to make this point clear